src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -229,7 +237,17 @@ export class NodeClientRequest extends ClientRequest {
       const mockedResponse = resolverResult.data
 
       if (mockedResponse) {
-        this.logger.info('received mocked response:', mockedResponse)
+        this.logger.info(
+          'received mocked response:',
+          mockedResponse.status,
+          mockedResponse.statusText
+        )
+
+        /**
+         * @note Ignore this request being destroyed by TLS in Node.js
+         * due to connection errors.
+         */
+        this.destroyed = false
```

Comment:
Instead of ignoring the failed request, can we handle the error by monkey-patching whatever is needed? This may not be urgent, and the issue may be resolved when/if we implement mocking at the socket level.
(https://github.com/mswjs/interceptors/pull/375)

Reply:
I'm not sure what we could monkey-patch in this case though. 

This assignment happens only for mocked responses. When you have a mock response returned, it shouldn't matter if there's been connection errors or not. This way we can keep the original connection logic from Node (handy for when you don't have mocked responses) but ignore any errors that occurred because they are irrelevant when providing a mock response. 

Note that we still do the following:

1. Capture any errors occurring during the request, which includes connection errors;
2. If the consumer hasn't responded to the request, we _replay those errors_ so they are never ignored when the request was performed as-is.

Reply:
> This assignment happens only for mocked responses...

I'm not an expert, but what if you want mock connection-related behavior, for example, Nock lets you [mock delay in the connection](https://github.com/nock/nock/#delay-the-connection) and mock delay in the response. 

Anyway, this is another feature or something anyway, so LGTM.

Reply:
Having more fine-grained delays would certainly be a feature worth exploring. For now, whichever delays you specify in the request listener are treated as respond delays (they trigger _after_ the connection has been made and the request has been written).

Reply:
Thanks! 
We will probably need to to keep parity with what Nock have now if we want to move Nock to `mswjs/interceptors`

Reply:
Let's raise this as a new feature request and discuss it in more detail.